src/services/services.ts

```diff
@@ -1519,11 +1520,15 @@ namespace ts {
             }
             const starts = getLineStarts(mappedFile);
             for (let index = starts.length - 1; index >= 0; index--) {
-                sourceMapCommentRegExp.lastIndex = starts[index];
-                const comment = sourceMapCommentRegExp.exec(mappedFile.text);
+                const lineText = mappedFile.text.substring(starts[index], starts[index + 1]);
+                const comment = sourceMapCommentRegExp.exec(lineText);
                 if (comment) {
                     return comment[1];
                 }
+                // If we see a nonwhitespace line, break, to avoid scanning up the entire file
+                else if (!lineText.match(whitespaceRegExp)) {

```

Comment:
The only other comment allowed after `//# sourceMappingURL` is `//# sourceURL`, which also needs to be skipped.